feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/presentation/staking/delegation/proxy/set/AddStakingProxyViewModel.kt

```diff
@@ -0,0 +1,233 @@
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+
+class AddStakingProxyViewModel(
+    addressInputMixinFactory: AddressInputMixinFactory,
+    feeLoaderMixinFactory: FeeLoaderMixin.Factory,
+    private val selectedAssetState: AnySelectedAssetOptionSharedState,
+    private val externalActions: ExternalActions.Presentation,
+    private val interactor: StakingInteractor,
+    private val accountRepository: AccountRepository,
+    private val assetUseCase: ArbitraryAssetUseCase,
+    private val resourceManager: ResourceManager,
+    private val selectAddressRequester: SelectAddressRequester,
+    private val addStakingProxyRepository: AddStakingProxyRepository,
+    private val validationExecutor: ValidationExecutor,
+    private val addStakingProxyValidationSystem: AddStakingProxyValidationSystem,
+    private val descriptionBottomSheetLauncher: DescriptionBottomSheetLauncher,
+    private val metaAccountGroupingInteractor: MetaAccountGroupingInteractor,
+) : BaseViewModel(),
+    DescriptionBottomSheetLauncher by descriptionBottomSheetLauncher,
+    ExternalActions by externalActions,
+    Validatable by validationExecutor {
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private val commissionAssetFlow = selectedAssetState.selectedChainFlow()
+        .flatMapLatest { assetUseCase.assetFlow(it.id, it.commissionAsset.id) }
+        .shareInBackground()
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private val selectedAssetFlow = selectedAssetState.selectedAssetFlow()
+        .flatMapLatest { assetUseCase.assetFlow(it) }
+        .shareInBackground()
+
+    val titleFlow = selectedAssetFlow.map {
+        resourceManager.getString(R.string.fragment_set_staking_proxy_title, it.token.configuration.symbol)
+    }
+        .shareInBackground()
+
+    val addressInputMixin = with(addressInputMixinFactory) {
+        val inputSpec = singleChainInputSpec(selectedAssetState.selectedChainFlow())
+
+        create(
+            inputSpecProvider = inputSpec,
+            myselfBehaviorProvider = noMyself(),
+            accountIdentifierProvider = web3nIdentifiers(
+                destinationChainFlow = selectedAssetState.assetWithChain,
+                inputSpecProvider = inputSpec,
+                coroutineScope = this@AddStakingProxyViewModel,
+            ),
+            errorDisplayer = this@AddStakingProxyViewModel::showError,
+            showAccountEvent = this@AddStakingProxyViewModel::showAccountDetails,
+            coroutineScope = this@AddStakingProxyViewModel,
+        )
+    }
+
+    val isSelectAddressAvailable = flowOf {
+        val selectedMetaAccount = accountRepository.getSelectedMetaAccount()
+        val chain = selectedAssetState.chain()
+        val filter = metaAccountsFilter(chain, selectedMetaAccount.requireAccountIdIn(chain))
+        metaAccountGroupingInteractor.hasAvailableMetaAccountsForChain(selectedAssetState.chainId(), filter)
+    }
+        .shareInBackground()
```

Comment:
Looks like "select address" could be its own mixin since this logic dublicates twice